app/javascript/mastodon/features/compose/util/language_detection_with_languagedetector.js

```diff
@@ -0,0 +1,41 @@
+import { debounce } from 'lodash';
+
+import { urlRegex } from './url_regex';
+
+const guessLanguage = async (text) => {
+  text = text
+    .replace(urlRegex, '')
+    .replace(/(^|[^/\w])@(([a-z0-9_]+)@[a-z0-9.-]+[a-z0-9]+)/ig, '');
+
+  try {
+    const languageDetector = await globalThis.LanguageDetector.create();
+    let {detectedLanguage, confidence} = (await languageDetector.detect(text))[0];
+    if (confidence > 0.8) {
+      detectedLanguage = detectedLanguage.split('-')[0];
+      return detectedLanguage;
+    }
+  } catch {
+    return '';
+  }
+
+  return '';
+};
+
+const debouncedGuess = (() => {
+  let resolver = null;
+
+  const debounced = debounce((text) => {
+    const result = guessLanguage(text);
```

Comment:
`guessLanguage` is an async function. Shouldn't this be wrapped in `await`/`async`?

Reply:
The result is `await`'ed below, so this works.